### ISSUE TITLE
Support for django 3.0 

### ISSUE THREAD

**Issue by pupattan** [awesto/django-shop#810]
Please support for django 3.0 .
django-cms has already upgraded

**Comment by jrief** [comment]
Thanks for this request, it's on my TODO list anyway!
In upcoming version 1.2, I will replace Haystack against Elasticsearch-DSL. I will add Django-3 support there as well. Unless it's really simple to backport it to django-SHOP 1.1. Had no time yet to have a look at it.
Would you mind to wait a little bit, until that version is ready? As an alternative, please check the branch `releases/1.2` to get the latest development release.

**Comment by pupattan** [comment]
Any expected (approx) date ?

**Comment by jrief** [comment]
I just managed to get django-SHOP working with Django-3.0.

However, there are still a few issues to solve:
- cookiecutter-django-shop must be adopted
- I must release djangocms-cascade version 1.3
- I must release django-angular version 2.3
- documentation about search must be adopted to Elasticsearch.

If you want to help, it would be appreciated.

**Comment by beeduino** [comment]
I want to help with cookiecutter-django-shop

**Comment by jrief** [comment]
Please join the [chat room](https://gitter.im/awesto/django-shop?utm_source=share-link&utm_medium=link&utm_campaign=share-link), so that we can work on it together.

**Comment by beeduino** [comment]
I am there

**Comment by pupattan** [comment]
> I just managed to get django-SHOP working with Django-3.0.
> 
> However, there are still a few issues to solve:
> 
> * cookiecutter-django-shop must be adopted
> * I must release djangocms-cascade version 1.3
> * I must release django-angular version 2.3
> * documentation about search must be adopted to Elasticsearch.
> 
> If you want to help, it would be appreciated.

Can you tell me how to get this update ?

**Comment by jrief** [comment]
@pupattan 
> Can you tell me how to get this update ?

* You must upgrade to django-angular 2.3 and djangocms-cascade-1.3.
* You must replace haystack against elasticsearch-dsl and django-elasticsearch-dsl
* Use django-SHOP from github with branch `releases/1.2`

**Comment by beeduino** [comment]
One of the changes needed for updates in cookiecutter-django-shop
https://github.com/awesto/djangoshop-sendcloud/pull/4

**Comment by beeduino** [comment]
And this
https://github.com/divio/djangocms-text-ckeditor/pull/527

**Comment by beeduino** [comment]
@jrief are you about to address this also?
`--- a/shop/management/commands/shop.py
+++ b/shop/management/commands/shop.py
@@ -225,7 +225,7 @@ Usage:
 
         page_attributes = [
             # Menu Title, CMS-App-Hook or None, kwargs, Main Plugin, Plugin Context,
-            (("Search", 'CatalogSearchCMSApp', {'reverse_id': 'shop-search-product'}),
+            (("Search", 'CatalogSearchApp', {'reverse_id': 'shop-search-product'}),
`

**Comment by beeduino** [comment]
One of the last problem that I am facing right now:
When running 
 $ pipenv run ./manage.py initialize_shop_demo
After migrating db it starts to load data and fails with:
```
  File "/home/dimas/.virtualenvs/my-shop-p7eeYaPN/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: Problem installing fixture '/home/dimas/.virtualenvs/awesto/p/my-shop/workdir/fixtures/skeleton.json': Could not load cmsplugin_cascade.CascadeClipboard(pk=25): NOT NULL constraint failed: cmsplugin_cascade_cascadeclipboard.created_at
```
I could overcome it adding null=True in the model but it looks hacky.

@jrief, can you take a look, please?

**Comment by jrief** [comment]
Yes, in djangocms-cascade I added that field. I forgot to add it to the file `skeleton.json` in cookiecutter-django-shop

**Comment by beeduino** [comment]
Still having some lesser problem when loading this skeleton.json: 
```shell
  Applying sites.0002_alter_domain_unique... OK
Initializing project myshop
You are trying to change the nullable field 'created_at' on cascadeclipboard to non-nullable without a default; we can't do that (the database needs something to populate existing rows).
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
...
Select an option: 1
Please enter the default value now, as valid Python
The datetime and django.utils.timezone modules are available, so you can do e.g. timezone.now
Type 'exit' to exit this prompt
>>> timezone.now
```

**Comment by beeduino** [comment]
Commented about cookiecutter-django-shop current status
https://github.com/awesto/cookiecutter-django-shop/pull/33#issuecomment-628135190

**Comment by pupattan** [comment]
I hope once this fix is done, we can use django-shop with django-cms 3.7.2

**Comment by jrief** [comment]
Today as of 20200517T0700 UTC I installed django-shop from scratch, using the cookiecutter template and following the given instructions. In my setup everything worked fine. If you still encounter problems, please send me you output.

**Comment by pupattan** [comment]
How do i integrate to existing django-cms project ?

**Comment by jrief** [comment]
This depends on how your current django-CMS is configured. If you add the django-SHOP specific configurations to your `settings.py`, then by invoking the management commands 
`./manage.py shop [check-pages|review-settings]` you can adopt the current content of a CMS to additionally use the Shop's funntionality.

**Comment by pupattan** [comment]
@jrief  Do you have any document so that i can refer ? """How do i integrate to existing django-cms project ?"""

**Comment by remoover** [comment]
PLEASE SOLVE IT .. WE NEED ALSO.. THANK YOU --> URGENT

**Comment by jrief** [comment]
@pupattan The `Pipfile` as provided by the cookiecutter template referrs to the external dependency `djangocms-cascade`. So if that's missing, try to reinstall it.

@remoover When reporting bugs, please do so in a reproducible manner.

> PLEASE SOLVE IT .. WE NEED ALSO.. THANK YOU --> URGENT

is not very informative to me.

**Comment by pupattan** [comment]
@jrief  Thanks for the info.
Could you please tell me how to integrate simply to existing CMS project (Python 3.8, django-cms 3.7.2, django 3.0,x)?
I found the following but looks very few info. 
https://django-shop.readthedocs.io/en/latest/features.html#django-shop-directly-plugs-into-django-cms
I am ok with few tweaks if some support issue (deprecated usages) are there.
